### PR TITLE
Remove biased annotations from prometheus.service.annotations

### PR DESCRIPTION
**Pull request by den-is**
#### Why is this pull request needed and what does it do?
Makes the default prometheus.service.annotations blank `{}` map. Removing any bias.
#### Which issues (if any) are related?
Fixes https://github.com/coredns/helm/issues/157

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/coredns/helm/blob/master/CONTRIBUTING.md#versioning).
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/coredns/helm/blob/master/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/coredns/helm/blob/master/CONTRIBUTING.md#developer-certificate-of-origin).

Changes are automatically published when merged to `main`. They are not published on branches.

<details>
  <summary>Note on DCO</summary>

  If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

### CODE REVIEW

**Comment by hagaibarel** [comment]
This would potentially break exiting installation that rely on this values being set by default

**Comment by den-is** [comment]
@hagaibarel correct.
well, c'est la vie.
But this is easy to fix for anyone.
Currently I'm kinda fixing my setup for myself too, by providing `null` and I could ignore the "issues" completely.
I just wanted to make this chart for the community better.
Otherwise for how long are you going to haul this """not-that-smart""" non-standard piece in values.yaml?

**Comment by hagaibarel** [comment]
> c'est la vie 

This isn't my preferred way of maintaing an OSS project where we have a community and users to support. If we wanted to introduce a change that would break people's exiting installation, this should have been a major version release, not a patch (or even a minor).

> Otherwise for how long are you going to haul this """not-that-smart""" non-standard piece in values.yaml?

Given that there isn't a "standard" in values file, it's common to provide sane defaults with an option to override / change them. 

Bottom line, I don't think this change has any meaningful value

**Comment by den-is** [comment]
added comment to the issue. I can bump chart version to 2.0.0

**Comment by hagaibarel** [comment]
I don't think this change justifies bumping a major version, nor has any significant value.

There is an option to override the annotations if needed, and the defaults make sense and fit most use cases.

**Comment by den-is** [comment]
The override option is not functioning 100% as one would think by default and is not covering all cases.

**Comment by hagaibarel** [comment]
Can you provide a real world use case from your experience where overriding didn't address the issue?

**Comment by den-is** [comment]
> Can you provide a real world use case from your experience where overriding didn't address the issue?

I have provided exact example in the original issue linked to this PR.
Real world **fact** is that maps in Helm are not overridden but merged.
